### PR TITLE
MGDOBR-637: fix openapi catalog schema

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-637](https://issues.redhat.com/browse/MGDOBR-637)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by jomarko — APPROVED** [review]
Approving the current state. However, for a record, we are solving here just the first issue mentioned in comment

https://issues.redhat.com/browse/MGDOBR-637?focusedCommentId=20361075&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-20361075

The second issue needs to be solved separately.

**Review by jomarko — APPROVED** [review]
Thank you @r00ta

File: infra/src/main/java/com/redhat/service/smartevents/infra/models/gateways/Gateway.java [hunk 1]
```diff
@@ -36,9 +37,14 @@ public class Gateway {
...
     @JsonProperty("parameters")
     // ObjectNode is not rendered properly by swagger
-    @Schema(implementation = Object.class, required = true)
+    @Schema(hidden = true)
     private ObjectNode parameters;
 
+    @JsonIgnore
+    // Workaround to generate the openapi.yaml file with a Map<String, Object> for the parameters. TODO: https://issues.redhat.com/browse/MGDOBR-788
+    @Schema(name = "parameters", required = true)
+    private Map<String, Object> justforopenapi;
```
Comment (jomarko) [inline infra/src/main/java/com/redhat/service/smartevents/infra/models/gateways/Gateway.java:46]: @r00ta after some discussion with @kelvah , we agreed changes in `Gateway.java` are not needed for now. Culd you please revert them?

**Review by jomarko — CHANGES_REQUESTED** [review]

File: infra/src/main/java/com/redhat/service/smartevents/infra/models/gateways/Gateway.java (hunk 1 above)
Comment (r00ta) [reply infra/src/main/java/com/redhat/service/smartevents/infra/models/gateways/Gateway.java:46]: I'll close this PR then

File: openapi/rhose-api.yaml [hunk 2]
```diff
@@ -592,7 +592,7 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/JsonNode'
+                type: object
```
Comment (kelvah) [inline openapi/rhose-api.yaml:595]: @r00ta Is there a way to keep this `object` declaration for the schema type without having also
```
additionalProperties:
            type: object
```
for the `parameters` definitions (lines 739 and 1038)?

The `additionalProperties` is generating a `{ [key:string]: object}` for parameters in the sdk. It's better to have them declared just as `object`, like before.

Comment (kelvah) [reply openapi/rhose-api.yaml:595]: @jomarko yes, the parameters are fine as they are. I was trying to understand if we can keep also the `type: object` on the schema definition.

Comment (r00ta) [reply openapi/rhose-api.yaml:595]: I do apologize, I forgot this PR included also this fix. I've reopened it :+1:

**Review by kelvah — APPROVED** [review]
Looking good. Thanks @r00ta :+1:

**Review by jomarko — APPROVED** [review]
Thank you @r00ta

**Pull request merged**